### PR TITLE
[develop2] transitivity fixes in PkgConfigDeps for components

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: PkgConfiDeps is using the wrong ``dependencies.host`` from dependencies instead of ``get_transitive_requires()`` computation.

### CODE REVIEW

File: conan/tools/gnu/pkgconfigdeps.py [hunk 1]
```diff
@@ -226,13 +228,8 @@ class PkgConfigConan(ConanFile):
...
             def package_info(self):
                 self.cpp_info.requires = ["other::cmp1"]
-        ```
-        Or:
 
-        ```python
-        from conan import ConanFile
-        class PkgConfigConan(ConanFile):
-            requires = "other/1.0"
+            # Or:
```
Comment (franramirez688) [inline conan/tools/gnu/pkgconfigdeps.py:232]: Why has it been removed?

**Review by franramirez688 — APPROVED** [review]
Looks good! FYI, `XcodeDeps` and `BazelDeps` are also using this `dependencies.direct_host` so I guess that we will have to change them as well.

**Review by czoido — APPROVED** [review]

File: conan/tools/gnu/pkgconfigdeps.py (hunk 1 above)
Comment (franramirez688) [reply conan/tools/gnu/pkgconfigdeps.py:232]: Oh, ok, I saw it 😅  I preferred to duplicate the example instead of this, but I think it's OK as well

Comment (memsharded) [reply conan/tools/gnu/pkgconfigdeps.py:232]: Not removed, just compacted, reusing the same part above, and putting the OR inside the ``package_info()``, so it is easier to see what is the difference.

**Pull request merged**